### PR TITLE
:broom: Clean up the `kn` sub-package

### LINKED ISSUE #152: Make package naming more consistent with skipping `kn` in the sub-dirs
Hmm, the package [knative.dev/client-pkg/pkg/config](https://pkg.go.dev/knative.dev/client-pkg) already exists, as well. I believe, that's why I put it in here...

We should reconcile those two packages somehow. IMHO, we shouldn't be using the [kn sub-package](https://pkg.go.dev/knative.dev/client-pkg/pkg/kn), as all of those packages are kn related, obviously.

IDK, maybe we should dump the [kn sub-package](https://pkg.go.dev/knative.dev/client-pkg/pkg/kn), probably in a separate PR, to avoid confusion. But, that's a breaking change, so we should prepare migration as well.

WDYT on those two, already existing, `config` packages?

_Originally posted by @cardil in https://github.com/knative/client-pkg/pull/139#discussion_r1504910909_

---            

Per the conversation on the linked PR. Let's follow-up on the package name changes.

### PR DESCRIPTION
**Pull request by cardil**
# Changes

- :broom: Clean up the `kn` sub-package

/kind cleanup

Fixes #152 

**Release Note**

```release-note
The knative.dev/client-pkg/pkg/kn was ditributed across the parent package knative.dev/client-pkg/pkg
```

### CODE REVIEW

File: pkg/kn-source-pkg/pkg/factories/flags_factory.go
Comment (cardil) [inline pkg/kn-source-pkg/pkg/factories/flags_factory.go:1]: The [knative.dev/client-pkg/pkg/kn-source-pkg](https://pkg.go.dev/knative.dev/client-pkg/pkg/kn-source-pkg) sub-package seems to me to be the same story as the `kn` package.

It was introduced in https://github.com/knative/client-pkg/pull/74, but contains some weird stuff, that shouldn't be present in a library:

 * [a main() file](https://github.com/knative/client-pkg/blob/f17097100e20efa13c05984a515e2746a7b40ea6/pkg/kn-source-pkg/cmd/main.go)
 * [some hack scripts](https://github.com/knative/client-pkg/tree/f17097100e20efa13c05984a515e2746a7b40ea6/pkg/kn-source-pkg/hack)
 * [e2e tests for kn-source-plugin, but (what looks like) name-replaced](https://github.com/knative/client-pkg/blob/f17097100e20efa13c05984a515e2746a7b40ea6/pkg/kn-source-pkg/test/e2e/basic_workflow_test.go)

I think we should clean up this as well, and leave just [the actual library](https://github.com/knative/client-pkg/tree/f17097100e20efa13c05984a515e2746a7b40ea6/pkg/kn-source-pkg/pkg), just renamed to `knative.dev/client-pkg/plugin/source/...`.

/cc @vyasgun 
/cc @dsimansk

Comment (cardil) [reply pkg/kn-source-pkg/pkg/factories/flags_factory.go:1]: /hold

Comment (dsimansk) [reply pkg/kn-source-pkg/pkg/factories/flags_factory.go:1]: ACK agreed on the proposed cleanup. IMO it's a leftover from the initial move of the library from the original repository into here.

File: pkg/kn/plugin/types.go
[the remark this answers is not in the document]
Comment (dsimansk) [reply pkg/kn/plugin/types.go:1]: ok

**Review by dsimansk — APPROVED** [review]

**Review by vyasgun — APPROVED** [review]

File: pkg/kn-source-pkg/pkg/factories/flags_factory.go
Comment (vyasgun) [reply pkg/kn-source-pkg/pkg/factories/flags_factory.go:1]: yes, everything including the tests was moved from the original repository and added to this one as a package

Comment (cardil) [reply pkg/kn-source-pkg/pkg/factories/flags_factory.go:1]: @dsimansk proposed to do this cleanup in a separate PR.

/unhold

Comment (cardil) [reply pkg/kn-source-pkg/pkg/factories/flags_factory.go:1]: I opened https://github.com/knative/client-pkg/issues/157 for this

**Pull request merged**